core.py

```diff
@@ -140,66 +121,61 @@ def process_order(self, cust_id):
                 food_qty = input("Enter food quantity: ")
-                self.customer.update_food_to_order(o.order_id, food_id, food_qty, session)
-                print("Items {}".format("updated" if True else "not updated"))
+                u = self.customer.update_food_to_order(session, o.order_id, food_id, food_qty)
+                print("Items {}".format("updated" if u else "not updated"))
 
             option = int(input(selection))
         
         if o:
-            print("Order {}. \nOrder number: {}".format("generated", o.order_id if True else "not generated"))
-
+            print("Order {}. \nOrder number: {}".format("generated", o.order_id if o else "not generated"))
 
-    def view_order(self, order_id):
-        session = self.Session()
-        view_order_item = self.customer.view_order_per_item(order_id, session)
+    @handle_session
+    def view_order(self, session, order_id):
+        view_order_item = self.customer.view_order_per_item(session, order_id)
         if view_order_item:
             for fc, fd, cos in view_order_item:
                 print("\nFood category: {} \nFood name: {} \nFood price: {} \nFood quantity: {} \nTotal per item: {}".format(
                     fc.name, fd.food_name, fd.price, cos.food_qty, (fd.price*cos.food_qty)))
 
-        view_order_grand = self.customer.view_order_grand_total(order_id, session)
+        view_order_grand = self.customer.view_order_grand_total(session, order_id)
         if view_order_grand:
             for fd, cos, cd, cosa, t in view_order_grand:
                 print("\nCustomer name: {} \nOrder ID: {} \nTotal bill: {}".format(cd.cust_name, cosa.order_id, t))
 
-
-    def checkout(self, order_id, order_status, order_address, checkout_time, estimated_time, bill_amount):
-        session = self.Session()
-        c = self.customer.checkout(order_id, order_status, order_address, 
-                        checkout_time, estimated_time, bill_amount, session) 
+    @handle_session
+    def checkout(self, session, order_id, order_status, order_address, checkout_time, estimated_time, bill_amount):
+        c = self.customer.checkout(session, order_id, order_status, order_address, 
+                        checkout_time, estimated_time, bill_amount) 
         if c:
-            print("Checkout {}!".format("successful" if True else "not successful"))
+            print("Checkout {}!".format("successful" if c else "not successful"))
 
-
-    def cancel_order(self, order_id, order_status):
-        session = self.Session()
-        c = self.customer.cancel_order(order_id, order_status, session) 
+    @handle_session
+    def cancel_order(self, session, order_id, order_status):
+        c = self.customer.cancel_order(session, order_id, order_status) 
         if c:
-            print("Cancel {}!".format("successful" if True else "not successful"))
+            print("Cancel {}!".format("successful" if c else "not successful"))
```

Comment:
Again, you're checking `if c` twice. It's wrong.

Reply:
change made